Issuetracker : #iXXXXX#
CWS          : N/A
Author       : <maho@openoffice.org> (JCA)
Description  : FreeBSD porting : building sdk fails for FreeBSD

1. Build error
dmake sdkoo fails like:

##############################################################################
... starting unpatched epm ...
... epm -f bsd openoffice.org-sdk /work/ports/editors/openoffice.org-2.0-devel/work/instsetoo_native/unxfbsd.pro/OpenOffice_SDK/bsd/listfile/en-US/epm_OpenOffice_SDK_gid_Module_Root_unxfbsd_SRC680__en-US.lst -v  2>&1 | ...

**************************************************
ERROR: ERROR: "epm -f bsd openoffice.org-sdk /work/ports/editors/openoffice.org-2.0-devel/work/instsetoo_native/unxfbsd.pro/OpenOffice_SDK/bsd/listfile/en-US/epm_OpenOffice_SDK_gid_Module_Root_unxfbsd_SRC680__en-US.lst -v  2>&1 |"!
in function: call_epm
**************************************************

**************************************************
ERROR: Saved logfile: /work/ports/editors/openoffice.org-2.0-devel/work/instsetoo_native/unxfbsd.pro/OpenOffice_SDK/bsd/logging/en-US/log_SRC680__en-US.log
**************************************************
... cleaning the output tree ...
... removing directory /work/ports/editors/openoffice.org-2.0-devel/work/instsetoo_native/unxfbsd.pro/OpenOffice_SDK/bsd/zip/en-US ...
Fri Nov 18 22:49:33 2005 (08:05 min.)
dmake:  Error code 255, while making 'sdkoo_en-US.bsd'
'---* tg_merge.mk *---'
##############################################################################

2. Reason why it fails:
since quoting for filename containing `$' produces an error

tar: opt/openoffice.org2.0_sdk/classes/com/sun/star/lib/loader/InstallationFinder\$StreamGobbler.class: Could not stat: No such file or directory
tar: opt/openoffice.org2.0_sdk/classes/com/sun/star/lib/loader/Loader\$CustomURLClassLoader.class: Could not stat: No such file or directory

3. Solution

For FreeBSD, we don't need quoting. We use printf for files names
instead of using qprintf, a function that treats quoting string.

--- epm/epm-3.7.patch	Sat Nov 19 10:21:34 2005
+++ epm/epm-3.7.patch	Sat Nov 19 10:22:23 2005
@@ -586,3 +586,23 @@
   }
   
 --- 457,462 ----
+
+*** misc/epm-3.7/bsd.c	Wed Jan 15 02:05:01 2003
+--- misc/build/epm-3.7/bsd.c	Sat Jun  4 06:18:27 2005
+***************
+*** 234,240 ****
+        case 'c' :
+        case 'f' :
+        case 'l' :
+!           qprintf(fp, "%s\n", file->dst + 1);
+            break;
+      }
+    }
+--- 243,249 ----
+        case 'c' :
+        case 'f' :
+        case 'l' :
+!           fprintf(fp, "%s\n", file->dst + 1);
+            break;
+      }
+    }
